activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -467,6 +467,17 @@ class DatabaseTasksStructureLoadTest < ActiveRecord::TestCase
   end
 
   class DatabaseTasksCheckSchemaFileTest < ActiveRecord::TestCase
+    def setup
+      unless Rails.respond_to?(:root)
+        @defined_root = true
+        def Rails.root; :defined_root; end
```

Comment:
I think I'll just remove the stubbing stuff and write an integration test that goes through the rake task, like the repro I posted in the description.